### ISSUE TITLE
MD013 sentence-per-line: a sentence wrapped onto a link-only line is not joined (standalone-link exemption ends the paragraph)

### ISSUE THREAD

**Issue by maxencefaldor** [rvben/rumdl#853]
### rumdl version

0.2.64

### Installation method

Other (`uvx rumdl`, the PyPI wheel)

### Bug description

In `sentence-per-line` mode, a sentence that wraps onto a line holding only a link is not reported as one sentence across two lines, and `fmt` does not join it. The same sentence wrapped onto a line holding a bare word, a code span, or bold text is reported and joined.

Cause, as far as I can tell from the source at fc410f7: the standalone-link exemption from the line-length check, [`is_standalone_link_or_image_line`](https://github.com/rvben/rumdl/blob/fc410f77a2167e1229eebfcc8c6ccd269e818fe5/src/rules/md013_line_length/helpers.rs#L393), also ends the paragraph during sentence-per-line paragraph collection ([here](https://github.com/rvben/rumdl/blob/fc410f77a2167e1229eebfcc8c6ccd269e818fe5/src/rules/md013_line_length.rs#L1553-L1559) and [here](https://github.com/rvben/rumdl/blob/fc410f77a2167e1229eebfcc8c6ccd269e818fe5/src/rules/md013_line_length.rs#L3371-L3377), both guarded by `!config.strict`). The exemption exists because a URL cannot be shortened, which is the right call for a width limit, but in sentence-per-line mode (and with `line-length = 0`, where no width applies at all) the line is a fragment of a sentence and should be absorbed into it. Consistent with this, `strict = true` makes the paragraph join.

### Steps to reproduce

1. `.rumdl.toml`:

   ```toml
   [MD013]
   line-length = 0
   reflow = true
   reflow-mode = "sentence-per-line"
   ```

2. Four one-paragraph files, each a single sentence over two lines:

   ```markdown
   A repository to train policies on
   [Craftax](https://example.com).
   ```

   ```markdown
   A repository to train policies on
   Craftax.
   ```

   ```markdown
   A repository to train policies on
   `craftax`.
   ```

   ```markdown
   A repository to train policies on
   [Craftax](https://example.com) and more.
   ```

3. `rumdl check` reports `Paragraph should have one sentence per line (found 1 sentences across 2 lines)` for the last three and nothing for the first. `rumdl fmt` joins the last three and leaves the first. Dropping the trailing period from the link line changes nothing.

4. `rumdl check -c 'MD013.strict=true'` on the first file reports the paragraph and joins it.

### Expected behavior

In `sentence-per-line` and `semantic-line-breaks` modes, a line whose only content is a link is treated as part of the sentence it continues, at least when `line-length = 0` (no width to protect). The width exemption would still apply to a link that is its own paragraph, which is the case it was written for.

**Comment by rvben** [comment]
Thanks @maxencefaldor. The standalone-link exemption was ending the paragraph in every mode, including the sentence modes, so a sentence that wrapped onto a link-only line never got joined.

That boundary now applies only in the width-driven modes, through a single predicate rather than the mode test repeated at five call sites. A link that is genuinely its own paragraph is still left alone. Fixed in v0.2.66.
